### PR TITLE
feat: Integrated statistics for #568

### PR DESCRIPTION
**Pull request by erickgirard**
Started to integrate the stats API in the SDK

**note**: this is a first commit, not yet ready for merge, mostly for code review from architecture standpoint

### CODE REVIEW

**Comment by thinkingserious** [comment]
@erickgirard

We have not been able to merge your Pull Request, but because you are awesome - we wanted to make sure you could still get a SendGrid Hacktoberfest shirt.  

Please go **fill [out our swag form](https://dx.sendgrid.com/swag) before Nov 5th** and we will send the shirt! (We know that you might have tried this before and it didn’t work, sorry about that!)

### **You have till Nov 5th to fill out this form** in order to get the Hacktoberfest shirt!

Thank you for contributing during Hacktoberfest! We hope to see you in the repos soon! Just so you know, we always give away a SendGrid shirt for your first ever non-Hacktoberfest PR that gets merged.

**Comment by erickgirard** [comment]
Hello @thinkingserious , FYI I still intend on finishing this PR and integrate all the stats, just wanted to make sure I'm on the rigth track before including the rest of the stats in the final PR.

**Comment by thinkingserious** [comment]
No problem @erickgirard,

We are currently working through the hacktoberfest PRs from oldest to newest. We amazingly had over 1,000 this October! So, we are a bit behind :(

Thank you for your support and patience!

With Best Regards,

Elmer

File: src/SendGrid/SendGridClient.cs [hunk 1]
```diff
@@ -46,6 +49,11 @@ public class SendGridClient : ISendGridClient
         private HttpClient client;
 
         /// <summary>
+        /// Acceptable format for date in the API
```
Comment (Jericho) [inline src/SendGrid/SendGridClient.cs]: This is one of the "acceptable format for date", but not the only one. There are at least three other formats that I'm aware are used by different endpoints in the SendGrid API (see [here](https://github.com/sendgrid/sendgrid-csharp/issues/193) for more details).

**Comment by thinkingserious** [comment]
Hello @erickgirard,

Just checking in to see if you wanted to continue work on this PR. If so, please address Jericho's comment and resolve the conflicts. Thank you!

With best regards,

Elmer

**Comment by thinkingserious** [comment]
Closing until PR feedback is addressed.

**Pull request closed without merging**